cluster/src/main/java/io/scalecube/cluster/membership/MembershipProtocolImpl.java

```diff
@@ -69,19 +72,14 @@
   private final Map<String, MembershipRecord> membershipTable = new HashMap<>();
 
   // Subject
-
-  private final Subject<MembershipEvent, MembershipEvent> subject =
-      PublishSubject.<MembershipEvent>create().toSerialized();
-
-  // Subscriptions
-
-  private Subscriber<Message> onSyncRequestSubscriber;
-  private Subscriber<Message> onSyncAckResponseSubscriber;
-  private Subscriber<FailureDetectorEvent> onFdEventSubscriber;
-  private Subscriber<Message> onGossipRequestSubscriber;
+  // Subject
+  private FluxProcessor<MembershipEvent, MembershipEvent> subject =
+      DirectProcessor.<MembershipEvent>create().serialize();
```

Comment:
Why here `.serialize` again, but at FailureDetector we dont have it.

Reply:
@artem-v  so what is you suggestion in such case?